src/components/testCaseLanding/qdm/TestCaseList.tsx

```diff
@@ -228,39 +229,65 @@ const TestCaseList = (props: TestCaseListProps) => {
   const clauseCoverageProcessor = (calculationOutput) => {
     //generates current populations coverage %
     if (calculationOutput && selectedPopCriteria) {
-      const trueClauses = new Set<string>();
-      const allClauses = new Set<string>();
+      let allClauses = [];
+      let relevantStatements;
       const patientIDs = Object.keys(calculationOutput);
       patientIDs.forEach((patientID) => {
-        if (
-          calculationOutput[patientID][selectedPopCriteria.id]
-            ?.clause_results?.[measure.cqlLibraryName]
-        ) {
-          const clauses =
+        let newClauseResults = [];
+        const clauseResults =
+          calculationOutput[patientID][selectedPopCriteria.id]?.clause_results;
+        if (clauseResults) {
+          newClauseResults = Object.values(
+            calculationOutput[patientID][selectedPopCriteria.id]?.clause_results
+          )?.flatMap(Object.values);
+        }
+        // we only need one copy of relevantStatements from the first group to match against
```

Comment:
```suggestion
        // we only need one copy of relevantStatements from the first patient of a selected group
```

Reply:
this is a utility function. we could move it to coverage builder. probably a tech debt.